typing/typemod.ml

```diff
@@ -150,14 +153,144 @@ let make p n i =
   let open Variance in
   set May_pos p (set May_neg n (set May_weak n (set Inj i null)))
 
+let rec iter_path_apply p ~f =
+  match p with
+  | Pident _ -> ()
+  | Pdot (p, _, _) -> iter_path_apply p ~f
+  | Papply (p1, p2) ->
+     iter_path_apply p1 ~f;
+     iter_path_apply p2 ~f;
+     f p1 p2 (* after recursing, so we know both paths are well typed *)
+
+let path_is_prefix =
+  let rec list_is_prefix ~strict l ~prefix =
+    match l, prefix with
```

Comment:
It seems that `strict` is always true at every call site. If so it should probably be removed.